service/history/api/multioperation/api.go

```diff
@@ -191,13 +202,20 @@ func Invoke(
 	}
 
 	// workflow hasn't been started yet: start and then apply update
-	resp, _, err := startAndUpdateWorkflow(
-		ctx,
-		shardContext,
-		workflowConsistencyChecker,
-		starter,
-		updater,
-	)
+	resp, outcome, err := startAndUpdateWorkflow(ctx, starter, updater)
+	if err == nil && outcome != startworkflow.StartNew {
```

Comment:
Naively one might think this condition would be hit if the conflict policy were `USE_EXISTING`. Can you confirm that's not possible / doesn't make sense?

Reply:
The block above this one, `currentWorkflowLease != nil` etc, should handle the case where there is a workflow already running and the request might or might not use the Use-Existing policy.

Once the execution gets here to `startAndUpdateWorkflow`, it expects no workflow with that ID running. But then - a lie - it is actually running (workflow locking ...). That's what this if block is for. In this case a `Use-Existing` (will translates to start outcome `StartReused`) is an error that requires a retry from the top. That's because the update was never applied.